tiled/server/dependencies.py

```diff
@@ -160,19 +166,16 @@ def expected_shape(
 
 
 def slice_(
-    slice: str = Query(None, pattern="^[-0-9,:]*$"),
+    slice: str = Query(None, pattern=SLICE_REGEX),
 ):
     "Specify and parse a block index parameter."
-    import numpy
-
-    # IMPORTANT We are eval-ing a user-provider string here so we need to be
-    # very careful about locking down what can be in it. The regex above
-    # excludes any letters or operators, so it is not possible to execute
-    # functions or expensive arithmetic.
-    return tuple(
-        [
-            eval(f"numpy.s_[{dim!s}]", {"numpy": numpy})
-            for dim in (slice or "").split(",")
-            if dim
-        ]
-    )
+
+    def np_style_slicer(indices: tuple):
```

Comment:
Might as well define these at global scope, rather than re-defining them on every call. Unless I'm missing something clever happening with closures here.